model-optimizer/extensions/middle/StridedSliceNormalizer_test.py

```diff
@@ -0,0 +1,231 @@
+"""
+ Copyright (C) 2018-2021 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import unittest
+
+import numpy as np
+
+from extensions.middle.StridedSliceNormalizer import StridedSliceNormalizer
+from extensions.ops.split import VariadicSplit
+from mo.front.common.partial_infer.concat import concat_infer
+from mo.front.common.partial_infer.utils import int64_array
+from mo.graph.graph import Node
+from mo.middle.passes.infer import partial_infer
+from mo.ops.strided_slice import StridedSlice
+from mo.utils.ir_engine.compare_graphs import compare_graphs
+from mo.utils.unittest.graph import build_graph, valued_const_with_data, regular_op_with_empty_data, \
+    connect, regular_op, empty_data
+
+# extended with existing concat
+
+edges = (
+    *connect('input', '0:strided_slice'),
+    *connect('begin', '1:strided_slice'),
+    *connect('end', '2:strided_slice'),
+    *connect('strides', '3:strided_slice'),
+    *connect('strided_slice', 'res')
+)
+
+
+class TestStridedSliceNormalizer(unittest.TestCase):
```

Comment:
I think tests with new_axis mask and ellipsis/extending should be added. 
And have you checked correctness of permutations (input rank <4, >=4 with new_axis and without new_axis) somewhere?

Reply:
I did manual "layer" tests with toy network and shape infer worked correctly as well ass permutation. Will add them to unittests.

Reply:
Some permutation unit-tests for 
1. `input_rank <4` and ellipsis and new_axis
2. `input_rank >=4` and shrink_axis mask 
3. `input_rank = 4` and shrink_axis and new_axis 

are here `model-optimizer/extensions/middle/StridedSliceNormalizer_test.py`. First two cases start from `test_strided_slice_permute_15` (line **780**)